oaknational/Oak-Web-Application
# Feat/new search hit card - LESQ-459

## Description

- Adds new search hit card
- Decouples search hits from lesson list and unit list (helpers/types)
- Decouple search tracking from components and add at page level
- Updates/adds tests
- Adds to storybook
- Adds new tag to search hits
- Fixes pagination padding 

Music year: {owa_music_year}

## Acceptance Criteria:

- [x]  New component for search
- [x]  New component shows all details on result (type, key stage, exam boards etc) (no exam or year data yet - LESQ-486)
- [x]  Lesson is pink circle, unit is blue square
- [x]  Units just have title, lessons have title and description
- [x]  See lesson/see unit button
- [x]  Whole thing clickable (for now)
- [x]  Lesson metadata component should be able to handle multiple exam boards
- [x]  search results page is using this new card

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2114--oak-web-application.netlify.thenational.academy/teachers/search?term=waves&flag=enable_2023_search_api

Check new and legacy hits take you to the correct page
Tracking searchOnClick event should be recorded 

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist
